examples/plugins/writing-plugins.rst

```diff
@@ -0,0 +1,131 @@
+       (e.g., the `PluginRunContextProvider class <https://github.com/mlflow/mlflow/blob/master/tests/resources/mlflow-test-plugin/mlflow_test_plugin/__init__.py#L23>`_
+       within the ``mlflow_test_plugin`` module) to register.
+     - `GitRunContext <https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/context/git_context.py#L36>`_,
+       `DefaultRunContext <https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/context/default_context.py#L41>`_
+
+   * - Plugins for overriding definitions of model registry APIs like mlflow.register_model.
+     - mlflow.model_registry_store
+     - **Note**: The model registry is in beta (as of MLflow 1.5), so APIs are not guaranteed to be stable & model-registry plugins may break in the
+       future.
+
+       The entry point value (e.g. ``mlflow_test_plugin:PluginRegistrySqlAlchemyStore``) specifies a custom subclass of
+       `mlflow.tracking.model_registry.AbstractStore <https://github.com/mlflow/mlflow/blob/master/mlflow/store/model_registry/abstract_store.py#L6>`_
+       (e.g., the `PluginRegistrySqlAlchemyStore class <https://github.com/mlflow/mlflow/blob/master/tests/resources/mlflow-test-plugin/mlflow_test_plugin/__init__.py#L33>`_
+       within the ``mlflow_test_plugin`` module)
+
+       The entry point name (e.g. ``file-plugin``) is the tracking URI scheme with which to associate the custom AbstractStore implementation.
+       In the example above, who install the plugin & set a tracking URI of the form "file-plugin://<path>" will use the custom AbstractStore
+       implementation defined in ``PluginFileStore``. The full tracking URI is passed to the ``PluginFileStore`` constructor.
+     - `SqlAlchemyStore <https://github.com/mlflow/mlflow/blob/master/mlflow/store/model_registry/sqlalchemy_store.py#L34>`_
+
+Testing Your Plugin
+~~~~~~~~~~~~~~~~~~~
+
+We recommend testing your plugin to ensure that it follows the contract expected by MLflow. For
+example, a tracking AbstractStore plugin should contain tests verifying correctness of its
+``log_metric``, ``log_param``, ... etc implementations. See also the tests for MLflow's
+reference implementations as an example:
+
+* `Example tracking AbstractStore tests <https://github.com/mlflow/mlflow/blob/master/tests/store/tracking/test_file_store.py>`_
+* `Example ArtifactRepository tests <https://github.com/mlflow/mlflow/blob/master/tests/store/artifact/test_local_artifact_repo.py>`_
+* `Example RunContextProvider tests <https://github.com/mlflow/mlflow/blob/master/tests/tracking/context/test_git_context.py>`_
+* `Example model-registry AbstractStore tests <https://github.com/mlflow/mlflow/blob/master/tests/store/model_registry/test_sqlalchemy_store.py>`_
+
+
+Distributing Your Plugin
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Assuming you've structured your plugin similarly to the example plugin, you can `distribute it
```

Comment:
Question here (cc @mateiz) - do we want to encourage people to notify us when they've written a plugin & maintain a list of known plugins somewhere in our docs?